src/pages/index.js

```diff
@@ -62,31 +101,99 @@ function displayImgModal(e) {
   imgFormModal.open();
 }
 
+function displayAvatarModal() {
+  cardFormValidator.toggleButtonState();
+  avatarFormModal.open();
+}
 // function that closes img display modal
 function handleCardClick(imageUrl, caption) {
   imgDisplayPopup.open(imageUrl, caption);
 }
 
 //function that saves profile modal inputs
 
-function handleProfileFormSubmit(formData) {
-  const { name, description } = formData;
-  userInfo.setUserInfo(name, description);
-  profileFormModal.close();
+function handleProfileFormSubmit({ name, description }) {
+  profileFormModal.setLoading(true);
+  api
+    .updateUserInfo(name, description)
+    .then(() => {
+      userInfo.setUserInfo(name, description);
+      profileFormModal.close();
+    })
+    .catch((error) => {
+      console.log(error);
+    })
+    .finally(() => {
+      profileFormModal.setLoading(false);
+    });
 }
 
 //function that save imgs modal inputs
 
-function handleImgFormSubmit(inputValues) {
-  const { name, link } = inputValues;
-  const newCard = createCard({ name, link });
-  section.addItem(newCard);
-  imgFormModal.close();
+function handleImgFormSubmit({ name, link }) {
+  api
```

Comment:
* According to the task you need to change submit buttons text (`Saving…`) while waiting for the server response. And the user should see it.